src/lib/local-storage-helpers.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Helper functions for loading and saving graphs to/from local storage
+ */
+
+/**
+ * takes a graph object & graph name, stringifies, and saves to local storage
+ * in "saved" object with unique key of date string
+ */
+export const saveGraphToLocal = (graph, name, preview, frames, frameIDs) => {
+  let savedGraphs = localStorage.getItem('gifsmos-saved-graphs');
+  if (!savedGraphs) {
+    savedGraphs = {
+      saved: {}
+    };
+  } else {
+    savedGraphs = JSON.parse(savedGraphs);
+  }
+  const newGraph = {
+    name,
+    graph,
+    preview,
+    frames,
+    frameIDs
+  };
+  let timeStamp = new Date().toString();
+  timeStamp = timeStamp.slice(0, timeStamp.indexOf('(') - 1);
+  savedGraphs.saved[timeStamp] = newGraph;
+  localStorage.setItem('gifsmos-saved-graphs', JSON.stringify(savedGraphs));
+};
+
+/**
+ * takes a date string and retrieves the graph object from local storage
+ * returns graph object
+ */
+
+export const getGraphFromLocal = dateString => {
+  let savedGraphs = JSON.parse(localStorage.getItem('gifsmos-saved-graphs'));
+  return savedGraphs.saved[dateString];
+};
+
+/**
+ * returns an array of objects with a name and dateStrings or
+ * null if none saved
+ */
+export const getSavedGraphsList = () => {
+  let savedGraphs = JSON.parse(localStorage.getItem('gifsmos-saved-graphs'));
+
+  return savedGraphs
+    ? Object.entries(savedGraphs.saved).map(([date, obj]) => [
+        date,
+        obj.name,
+        obj.preview
+      ])
+    : savedGraphs;
```

Comment:
Rather than returning savedGraphs, perhaps return `null` ? If the key is missing, the value will be null, anyway, but it's easier for the reader to intuit what is being changed if no such key exists.

Reply:
done